kinto_client/__init__.py

```diff
@@ -117,265 +110,113 @@ def request(self, method, url, data=None, permissions=None, **kwargs):
+        resp, _ = self.session.request('put', endpoint, data=data,
+                                       permissions=permissions)
+        return resp
+
+    def update_record(self, data, collection=None, permissions=None,
+                      bucket=None):
+        # XXX How should we deal with permissions? Should a dict with the
+        # data and permissions keys be passed?
+        endpoint = self._get_endpoint('record', bucket, collection,
+                                      data['id'])
+        return self.session.request('put', endpoint, data=data,
+                                    permissions=permissions)
+
+    def delete_record(self, id, collection=None, bucket=None):
+        endpoint = self._get_endpoint('record', bucket, collection, id)
+        resp, _ = self.session.request('delete', endpoint)
+        return resp['data']
 
-class Record(object):
-    """Represents a record"""
-
-    def __init__(self, data=None, collection=None, bucket='default',
-                 permissions=None, server_url=None, auth=None, session=None,
-                 id=None, create=False, load=True):
-        self.session = create_session(server_url, auth, session)
-        if collection is None:
-            raise AttributeError("collection is mandatory")
-        if isinstance(collection, six.string_types):
-            collection = Collection(
-                collection,
-                bucket=bucket,
-                session=self.session,
-                load=False)
-        self.collection = collection
-
-        if id is None:
-            if data and ID_FIELD in data:
-                id = data[ID_FIELD]
-            else:
-                # If no id is specified, generate a new one.
-                id = str(uuid.uuid4())
-        self.id = id
-        self.data = data
-
-        self.permissions = Permissions('record', permissions)
-        self.uri = "%s/records/%s" % (self.collection.uri, self.id)
-
-        if load:
-            # XXX put in a different method.
-            method = 'put' if create else 'get'
-            request_kwargs = {}
-            if create:
-                request_kwargs['data'] = data
-                if permissions is not None:
-                    request_kwargs['permissions'] = permissions
-
-            info, _ = self.session.request(method, self.uri, **request_kwargs)
-            self.data = info['data']
-            self.permissions = Permissions('record', info['permissions'])
-
-        self.last_modified = self.data.pop('last_modified', None)
-        self.data.pop(ID_FIELD, None)
-
-    def save(self):
-        # XXX Rename in update_record and do a PATCH ?
-        self.session.request('put', self.uri,
-                             data=self.data,
-                             permissions=self.permissions)
-
-    def delete(self):
-        self.session.request('delete', self.uri)
+    def delete_records(self, records):
+        # XXX To be done with a BATCH operation
```

Comment:
Do you plan to do a context manager like redis pipeline? ``with client.pipeline as multi:``

The dream code could be:

```
    with self.batch() as batched:
        for record in records:
            batched.delete(record_id, bucket, collection)
        batched.execute()
```